lib/paratii.eth.wallet.js

```diff
@@ -15,21 +15,31 @@ export function patchWallet (wallet, config) {
       if (this._mnemonic) {
         mnemonic = this._mnemonic
       } else {
-        mnemonic = newMnemonic()
+        mnemonic = bip39.generateMnemonic()
       }
     }
-    if (isValidMnemonic(mnemonic)) {
+    if (bip39.validateMnemonic(mnemonic)) {
       this._mnemonic = mnemonic
       this._hdIndex = 0
-      // this code is lifted from eth-lightwallet
-      var hdRoot = new Mnemonic(this._mnemonic).toHDPrivateKey().xprivkey
+
+      var seed = bip39.mnemonicToSeed(this._mnemonic, this._passphrase)
+      // saving for testing purpose
+      this._seedHex = bip39.mnemonicToSeedHex(this._mnemonic, this._passphrase)
+
+      // contains masternode extended priv key and extended pub key
+      var masternode = hdkey.fromMasterSeed(seed)
+
+      // saving xtended private key for testing purpose
+      this._xpriv = masternode.privateExtendedKey()
 
       // var keys = []
       for (var i = 0; i < numberOfAccounts; ++i) {
-        var hdprivkey = new bitcore.HDPrivateKey(hdRoot).derive(this._hdIndex++)
-        var privkeyBuf = hdprivkey.privateKey.toBuffer()
+        var generatedAddress = masternode.deriveChild(i).getWallet()
+
+        var privkeyBuf = generatedAddress.getPrivateKey()
+
+        var privkeyHex = generatedAddress.getPrivateKey().toString('hex')
 
-        var privkeyHex = privkeyBuf.toString('hex')
         if (privkeyBuf.length < 32) {
```

Comment:
we do not need thesse checks anymore (they were a workaround around the buggy lib we used before). Can you remove them?